src/main/scala/temple/ast/Metadata.scala

```diff
@@ -7,10 +7,11 @@ import temple.errors.ErrorHandlingContext
 sealed trait Metadata
 
 object Metadata {
-  sealed trait TargetMetadata  extends Metadata
-  sealed trait ServiceMetadata extends Metadata
-  sealed trait ProjectMetadata extends Metadata
-  sealed trait StructMetadata  extends Metadata
+  sealed trait TargetMetadata          extends Metadata
+  sealed trait ProjectMetadata         extends Metadata
+  sealed trait ServiceOrStructMetadata extends Metadata
+  sealed trait ServiceMetadata         extends ServiceOrStructMetadata
+  sealed trait StructMetadata          extends ServiceOrStructMetadata
```

Comment:
This is done so that `temple.builder.project.ProjectBuilder.endpoints` works with structs. Would love a better solution!

Reply:
Union types <3

Reply:
https://dcapwell.github.io/scala-tour/Union%20and%20Intersection%20types.html

Reply:
DeMorgan's Types <3